python/common/business-registry-model/pyproject.toml

```diff
@@ -33,10 +33,168 @@ freezegun = "^1.5.1"
 pytest = "^8.3.5"
 zimports = "^0.6.1"
 black = "^25.1.0"
+pytest-cov = "^6.0.0"
 
+[tool.ruff]
+# Exclude a variety of commonly ignored directories.
+exclude = [
+    ".vscode",
+    "dist",
+    "htmlcov",
+    "venv",
+    "migrations",
+    "devops",
+    ".history"
+]
 
 [tool.poetry.dependencies]
 sql-versioning = {develop = true}
 [build-system]
 requires = ["poetry-core>=2.0.0,<3.0.0"]
 build-backend = "poetry.core.masonry.api"
+
+
+# Same as Black.
+line-length = 120
+indent-width = 4
+
+# Assume Python 3.12
+#target-version = "py312"
+
+[tool.ruff.format]
+# Like Black, use double quotes for strings.
+quote-style = "double"
+
+# Like Black, indent with spaces, rather than tabs.
+indent-style = "space"
+
+# Like Black, respect magic trailing commas.
+skip-magic-trailing-comma = false
+
+# Like Black, automatically detect the appropriate line ending.
+line-ending = "auto"
+
+# Enable auto-formatting of code examples in docstrings. Markdown,
+# reStructuredText code/literal blocks and doctests are all supported.
+#
+# This is currently disabled by default, but it is planned for this
+# to be opt-out in the future.
+docstring-code-format = true
+
+# Set the line length limit used when formatting code snippets in
+# docstrings.
+#
+# This only has an effect when the `docstring-code-format` setting is
+# enabled.
+docstring-code-line-length = "dynamic"
+
+[tool.ruff.lint]
+select = [
+    "B",   # flake8-bugbear
+    "C4",  # flake8-comprehensions
+    "ERA", # flake8-eradicate/eradicate
+    "I",   # isort
+    "N",   # pep8-naming
+    "PIE", # flake8-pie
+    "PL",  # pyLint
+    "PGH", # pygrep
+    "RUF", # ruff checks
+    "SIM", # flake8-simplify
+    "T20", # flake8-print
+    "TCH", # flake8-type-checking
+    "TID", # flake8-tidy-imports
+    "UP",  # pyupgrade
+]
+#ignore = [
+#    "B904", # use 'raise ... from err'
+#    "B905", # use explicit 'strict=' parameter with 'zip()'
+#    "N999"
+#]
+[tool.ruff.lint.per-file-ignores]
```

Comment:
I think its a good idea to have a comment for each rule or each group so that we can work towards unignoring some of these (particularly for the business_model grp) or maybe a tech debt ticket created to clean these up with the number commented here. Otherwise I think people might copy this config for new projects without realizing we actually prefer most of these as unignored